framework/db/mssql/QueryBuilder.php

```diff
@@ -174,31 +174,28 @@ public function renameColumn($table, $oldName, $newName)
      */
     public function alterColumn($table, $column, $type)
     {
-        $sqlAfter = [];
+        $sqlAfter[] = $this->dropConstraintsForColumn($table, $column, 'D');
 
-        $columnName = $this->db->quoteColumnName($column);
-        $tableName = $this->db->quoteTableName($table);
+        if ($type instanceof \yii\db\mssql\ColumnSchemaBuilder) {
+            $type->isAlterColumn();
 
-        $constraintBase = preg_replace('/[^a-z0-9_]/i', '', $table . '_' . $column);
+            $columnName = $this->db->quoteColumnName($column);
+            $tableName = $this->db->quoteTableName($table);
```

Comment:
It can be taken outside the `if` since it's used again in `return` statement. Also we can use it in `return`.